backend/ethereum/subscription/resistanteventsub_test.go

```diff
@@ -0,0 +1,143 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package subscription_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+
+	"perun.network/go-perun/backend/ethereum/bindings/peruntoken"
+	"perun.network/go-perun/backend/ethereum/channel/test"
+	"perun.network/go-perun/backend/ethereum/subscription"
+	"perun.network/go-perun/log"
+	pctx "perun.network/go-perun/pkg/context"
+	pkgtest "perun.network/go-perun/pkg/test"
+)
+
+var event = func() *subscription.Event {
+	return &subscription.Event{
+		Name: "Approval",
+		Data: new(peruntoken.PerunTokenApproval),
+	}
+}
+
+// TestResistantEventSub_Confirmations tests that a TX is confirmed exactly
+// after being included in `confirmations` many blocks.
+func TestResistantEventSub_Confirmations(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+	rng := pkgtest.Prng(t)
+	require := require.New(t)
+	s := test.NewTokenSetup(ctx, t, rng)
+
+	for i := 0; i < 3; i++ {
+		finality := rng.Int31n(10) + 1
+		sub, err := subscription.Subscribe(ctx, s.CB, s.Contract, event, 0, uint64(finality))
+		require.NoError(err)
+		log.Debugf("Needed confirmations: %d", finality)
+		// Send and Confirm the TX. The simulated backend already mined a block here,
+		// so the TX has 1 confirmation now.
+		s.ConfirmTx(s.IncAllowance(ctx), true)
+		// Wait `finality-1` blocks.
+		for j := int32(0); j < finality-1; j++ {
+			NoEvent(require, sub)
+			log.Debug("Commit")
+			s.SB.Commit()
+		}
+		log.Debug("Waiting for confirm")
+		OneEvent(require, sub)
+		sub.Close()
+		s.SB.Commit() // Make sure the next EventSub does not get some old block.
+	}
+}
+
+// TestResistantEventSub_FinalityDepth tests that a TX is confirmed exactly
+// after `finalityDepth` blocks even when reorgs occur.
+func TestResistantEventSub_Reorg(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+	rng := pkgtest.Prng(t)
+	require := require.New(t)
+	s := test.NewTokenSetup(ctx, t, rng)
+
+	for i := 0; i < 3; i++ {
+		finality := rng.Int31n(20) + 2
+		sub, err := subscription.Subscribe(ctx, s.CB, s.Contract, event, 0, uint64(finality))
+		require.NoError(err)
+
+		for i := 0; i < 20; i++ {
+			s.SB.Commit()
+		}
+		// Send and Confirm the TX
+		s.IncAllowance(ctx)
+		// Reorg until the block hight hits `finality` blocks.
+		for h := int64(0); h < int64(finality-1); {
+			d := rng.Int63n(10) + 1
+			l := rng.Int63n(10) + 1
+			NoEvent(require, sub)
+			log.Debugf("[h=%d] Reorg with depth: %d, length: %d", h, d, l)
+			s.SB.Reorg(ctx, uint64(d), func(txs []types.Transactions) []types.Transactions {
+				return append(txs, make([]types.Transactions, int(d+l)-len(txs))...)
+			})
+			h += l
+		}
+		log.Debug("done")
+		// Verify that the event arrived.
+		OneEvent(require, sub)
+		sub.Close()
```

Comment:
Above, you add another `s.SB.Commit()` here. Looks like we don't need it? Or we should also add it here?

Reply:
I remove the loops, so it is superfluous now.